src/engine/component/Player.ts

```diff
@@ -1,5 +1,18 @@
-import { defineComponent, defineQuery } from "bitecs";
+import { defineComponent, defineQuery, enterQuery, exitQuery, Not } from "bitecs";
+
+import { Owned } from "../network";
 
 export const Player = defineComponent({});
 
 export const playerQuery = defineQuery([Player]);
+export const enteredPlayerQuery = enterQuery(playerQuery);
+export const exitedPlayerQuery = exitQuery(playerQuery);
+
+export const ownedPlayerQuery = defineQuery([Player, Owned]);
+export const enteredOwnedPlayerQuery = enterQuery(ownedPlayerQuery);
+export const exitedOwnedPlayerQuery = exitQuery(ownedPlayerQuery);
+
+// eslint-disable-next-line new-cap
```

Comment:
This is going to be annoying when using `Not()` or `Changed()`. I wonder if we can add a better exception for this?